mlir/lib/Dialect/Linalg/Utils/Utils.cpp

```diff
@@ -240,6 +240,608 @@ bool isReductionIterator(utils::IteratorType iteratorType) {
+template <>
+bool isaConvolutionOpOfType<linalg::DepthwiseConv1DNwcWcOp>(
+    LinalgOp op, SmallVector<int64_t> *dilations,
+    SmallVector<int64_t> *strides) {
+  if (isa<linalg::DepthwiseConv1DNwcWcOp>(op))
+    return true;
+
+  assert(isaConvolutionOpInterface(op) &&
+         "expected linalgOp to implement ConvolutionOpInterface");
+
+  ArrayAttr indexingMaps = op.getIndexingMaps();
+  if (!verifyConvIndexingMapSizes(indexingMaps, {3, 2, 3}))
+    return false;
+
+  Block *body = op.getBlock();
+  auto yieldOp = cast<linalg::YieldOp>(body->getTerminator());
+  Value yieldVal = yieldOp.getOperand(0);
+  unsigned inputMapIdx = 0, filterMapIdx = 1, outputMapIdx = 2;
+
+  SmallVector<int64_t> tempDilations(1, 1);
+  SmallVector<int64_t> tempStrides(1, 1);
+  // Match: N
+  if (getAffineMapDim(indexingMaps, inputMapIdx, 0) !=
+      getAffineMapDim(indexingMaps, outputMapIdx, 0))
+    return false;
+  // Match: C
+  if (getAffineMapDim(indexingMaps, inputMapIdx, 2) !=
+      getAffineMapDim(indexingMaps, filterMapIdx, 1))
+    return false;
+  if (getAffineMapDim(indexingMaps, inputMapIdx, 2) !=
+      getAffineMapDim(indexingMaps, outputMapIdx, 2))
+    return false;
+  // Match: W + w
+  if (!matchConvDimAddExprPattern(indexingMaps, /*iDim=*/1, /*fDim=*/0,
+                                  /*oDim=*/1, tempDilations[0], tempStrides[0]))
+    return false;
+  // Match body
+  if (!bodyMatcherForConvolutionOps(yieldVal, body))
+    return false;
+  *dilations = SmallVector<int64_t>(tempDilations);
+  *strides = SmallVector<int64_t>(tempStrides);
+  return true;
+}
+
+// #inputMap = affine_map<(N, H, W, C, h, w) -> (N, C, H + h, W + w)>
+// #filterMap = affine_map<(N, H, W, C, h, w) -> (C, h, w)>
+// #outputMap = affine_map<(N, H, W, C, h, w) -> (N, C, H, W)>
+template <>
+bool isaConvolutionOpOfType<linalg::DepthwiseConv2DNchwChwOp>(
+    LinalgOp op, SmallVector<int64_t> *dilations,
+    SmallVector<int64_t> *strides) {
+  if (isa<linalg::DepthwiseConv2DNchwChwOp>(op))
+    return true;
+
+  assert(isaConvolutionOpInterface(op) &&
+         "expected linalgOp to implement ConvolutionOpInterface");
+
+  ArrayAttr indexingMaps = op.getIndexingMaps();
+  if (!verifyConvIndexingMapSizes(indexingMaps, {4, 3, 4}))
+    return false;
+
+  Block *body = op.getBlock();
+  auto yieldOp = cast<linalg::YieldOp>(body->getTerminator());
+  Value yieldVal = yieldOp.getOperand(0);
+  unsigned inputMapIdx = 0, filterMapIdx = 1, outputMapIdx = 2;
+
+  SmallVector<int64_t> tempDilations(2, 1);
+  SmallVector<int64_t> tempStrides(2, 1);
+  // Match: N
+  if (getAffineMapDim(indexingMaps, inputMapIdx, 0) !=
+      getAffineMapDim(indexingMaps, outputMapIdx, 0))
+    return false;
+  // Match: C
+  if (getAffineMapDim(indexingMaps, inputMapIdx, 1) !=
+      getAffineMapDim(indexingMaps, filterMapIdx, 0))
+    return false;
+  if (getAffineMapDim(indexingMaps, inputMapIdx, 1) !=
+      getAffineMapDim(indexingMaps, outputMapIdx, 1))
+    return false;
```

Comment:
Shouldn't this also be matching `H` and `W`?  Also, why:
```cpp
  // Match: C
  if (getAffineMapDim(indexingMaps, inputMapIdx, 1) !=
      getAffineMapDim(indexingMaps, outputMapIdx, 1))
      return false;
```
instead of simply (similar number of characters, but iMHO the intent is clearer):
```cpp
  // Match: C
  auto inputMap = indexingMaps[0];
  auto outputMap = indexingMap[2];
  if (inputMap.getResult(1) != outputMap.getResult(1))
```

Related to that, why no construct the expected maps and do a map comparison, similarly to: https://github.com/llvm/llvm-project/blob/71b21b5391b267f10c8222fbbe73494fbce6dd86/mlir/lib/Dialect/Linalg/Transforms/Vectorization.cpp#L4089-L4091

In fact, you might be able to re-use that, no?

Reply:
> Were you suggesting to use StructuredGenerator for the above layout(...) API ?

That would probably be an overkill 😅 What you have implement is 100% sufficient.